### DISCUSSION TITLE
[Feature] Expose more substitution variables in storage templates.

### DISCUSSION THREAD

**Discussion by psyciknz** [immich-app/immich#9905] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Allow for the possibility for other substitution variables for storage templates.
Ideally my use case, Make
But equally  Model and/or any of the main set of exif values.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by gamerkonks** [comment]
I am also hoping for this functionality to be implemented.

**Comment by mouie** [comment]
+1 as I was also looking to submit this feature request

**Comment by fbennedetto** [comment]
I would like to add my support for this feature, but I'd like to add on to it as well.

I'd like to see the ability to add sequential numbering with customizable padding.

My photos are named like this (to use today's date as an example):

102624001.jpg
102624002.jpg
102624003.jpg

And so on. They're numbered in the order in which they were taken, so 102624001.jpg would be the first photo taken on 10/26/24.

I'd like the ability for Immich to name the files like this, basing the order in which the photos were taken off the timestamp in the EXIF data.

**Reply by bo0tzz** [reply]
As a hacky-ish workaround, if you just set the template to generate the same name for these files (ie 102624.jpg), Immich will add an increasing suffix like 102624+1.jpg. I don't know how happy that'd be with a bunch of files though, and the numbering will change depending on adding/removing assets - though that seems a given for this sort of system.

**Comment by simowilso** [comment]
I'd love to have camera model available as a variable for the storage template, this is one directory level for me in my current external library

**Comment by Lord-Quayla** [comment]
Yes would love to have camera Model as a variable. Currently trying to organise all of my photos by CameraModel/Year/Month/IMAGE_56437.jpg. I find it really helps particularly when I mange photos from multiple family members with different camera.

**Comment by chancedonmillion** [comment]
I too would like this feature.

**Comment by rahul-kumar-saini** [comment]
[PR](https://github.com/immich-app/immich/pull/24650) for this!

**Reply by Nriver** [reply]
Nice work!

**Reply by Lord-Quayla** [reply]
Thank you for your work on this!
